File: java/sevenwonders.java
import java.util.Scanner;

public class sevenwonders {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		String cards = scan.next();
		int T = 0, C = 0, G = 0;
		for (char c : cards.toCharArray())
			switch (c) {
			case 'T':
				T++;
				break;
			case 'C':
				C++;
				break;
			default:
				G++;
				break;
			}
		System.out.println(T * T + C * C + G * G + 7 * Math.min(Math.min(T, C), G));
		scan.close();
	}
}

File: ts/src/sevenwonders.ts
import readline = require('readline')

function getScore (cards: string[]): number {
  let T = 0; let C = 0; let G = 0
  cards.forEach(card => {
    if (card.startsWith('C')) {
      C++
    } else if (card.startsWith('T')) {
      T++
    } else if (card.startsWith('G')) {
      G++
    }
  })
  return T * T + C * C + G * G + 7 * Math.min(T, C, G)
}

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    console.log(getScore(line.split('')))
  })
  .on('close', () => {
    process.exit(0)
  })
